projects/ad9694_500ebz/common/ad9694_500ebz_qsys.tcl

```diff
@@ -54,6 +54,7 @@ add_connection sys_clk.clk axi_ad9694_core.s_axi_clock
 add_instance util_ad9694_cpack util_cpack
 set_instance_parameter_value util_ad9694_cpack {CHANNEL_DATA_WIDTH} $CHANNEL_DATA_WIDTH
 set_instance_parameter_value util_ad9694_cpack {NUM_OF_CHANNELS} $NUM_OF_CHANNELS
+set_instance_parameter_value util_ad9694_cpack {SAMPLE_WIDTH} $SAMPLE_RESOLUTION
```

Comment:
I think we're going to or could have issues with this approach. After axi_adc_jesd core each sample are scaled into 8 or 16 bits. But the sample resolution can be anything between 7 and 16. Also I doubt that the cpack module will work with a SAMPLE_WIDTH different than 8 or 16 bits. We can use 'octets per samples' instead of 'sample width' for cpack.

Reply:
makes sense, we should round up to the next multiple of 8.

I think the cpack doesn't care too much about the width. On the data path it is just the same structure over and over repeated for the number of data bits. And the control logic does not depend on the number of data bits.